brave_domains/service_domains.cc

```diff
@@ -59,42 +61,66 @@ void MaybeWarnSwitchValue(std::string key, std::string value) {
   }
 }
 
+#if !defined(OFFICIAL_BUILD)
+std::string ConvertEnvironmentToString(brave_domains::ServicesEnvironment env) {
+  static const base::flat_map<ServicesEnvironment, std::string> envMap = {
+      {brave_domains::ServicesEnvironment::DEV, kBraveServicesSwitchValueDev},
+      {brave_domains::ServicesEnvironment::STAGING,
+       kBraveServicesSwitchValueStaging},
+      {brave_domains::ServicesEnvironment::PROD,
+       kBraveServicesSwitchValueProduction}};
+
+  auto it = envMap.find(env);
+  if (it != envMap.end()) {
+    return it->second;
+  }
+
+  NOTREACHED();
+  return kBraveServicesSwitchValueProduction;
+}
+#endif
+
 }  // namespace
 
 std::string GetServicesDomain(
-    std::string prefix /* = "" */,
+    std::string prefix,
+    ServicesEnvironment env_value_default_override,
```

Comment:
specifying a default when you call `GetServicesDomain` seems extremely fragile/error prone to me and could result in different parts of a feature having different default envs in non official builds. My understanding is that the problem is that some services only have two environments in which case we should just map dev and staging to the same thing

Reply:
we can do it at the dns level